rest_framework/views.py

```diff
@@ -186,8 +187,13 @@ def get_authenticate_header(self, request):
         header to use for 401 responses, if any.
         """
         authenticators = self.get_authenticators()
+        www_authenticate_behavior = self.www_authenticate_behavior
         if authenticators:
-            return authenticators[0].authenticate_header(request)
+            if www_authenticate_behavior == 'first':
+                return authenticators[0].authenticate_header(request)
+            elif www_authenticate_behavior == 'all':
+                challenges = (a.authenticate_header(request) for a in authenticators)
+                return ', '.join((c for c in challenges if c is not None))
```

Comment:
```suggestion
                return ', '.join((c for c in challenges if c is not None))
            else:
                raise ValueError(f"Do not know how to handle {www_authenticate_behavior} www_authenticate_behavior. Use 'first' or 'all'")
```
or alike?

Reply:
Performing this check during "runtime" is probably not the right thing -- it would be better to validate the setting at startup, but I'm not sure if/how to do that. Perhaps advice from @auvipy could help us out.

Reply:
I added a Django system check for the setting in 525979e.